pages/index.js

```diff
@@ -1,7 +1,60 @@
-export default function HomePage() {
+import Head from "next/head";
+import React, { useRef, useEffect, useState } from "react";
+import "mapbox-gl/dist/mapbox-gl.css";
+import mapboxgl from "!mapbox-gl"; // eslint-disable-line import/no-webpack-loader-syntax
+
+export function getServerSideProps() {
+  return {
+    props: {
+      accessToken: process.env.MAPBOX_TOKEN,
+    },
+  };
+}
+
+export default function Home({ accessToken }) {
+  const mapContainer = useRef(null);
+  const map = useRef(null);
+  const [lng, setLng] = useState(-70.9);
+  const [lat, setLat] = useState(42.35);
+  const [zoom, setZoom] = useState(9);
+
+  mapboxgl.accessToken = accessToken;
+
+  useEffect(() => {
+    if (map.current) return; // initialize map only once
+    map.current = new mapboxgl.Map({
+      container: mapContainer.current,
+      style: "mapbox://styles/mapbox/streets-v12",
+      center: [lng, lat],
+      zoom: zoom,
+    });
+  });
+
+  useEffect(() => {
+    if (!map.current) return; // wait for map to initialize
+    map.current.on("move", () => {
+      setLng(map.current.getCenter().lng.toFixed(4));
+      setLat(map.current.getCenter().lat.toFixed(4));
+      setZoom(map.current.getZoom().toFixed(2));
+    });
+  });
```

Comment:
not sure about how the Library works, but would it still work if you wrote it like this:

```suggestion
  useEffect(() => {
    map.current.on("move", () => {
      setLng(map.current.getCenter().lng.toFixed(4));
      setLat(map.current.getCenter().lat.toFixed(4));
      setZoom(map.current.getZoom().toFixed(2));
    });
  }, []);
```